src/behaviors/humanoid/walk.cpp

```diff
@@ -0,0 +1,224 @@
+#include "inria_wbc/behaviors/humanoid/walk.hpp"
+
+namespace inria_wbc {
+    namespace behaviors {
+        namespace humanoid {
+            static Register<Walk> __walk("humanoid::walk");
+
+            Walk::Walk(const controller_ptr_t& controller, const YAML::Node& config) : Behavior(controller, config)
+            {
+                auto c = IWBC_CHECK(config["BEHAVIOR"]);
+                traj_com_duration_ = IWBC_CHECK(c["traj_com_duration"].as<float>());
+                traj_foot_duration_ = IWBC_CHECK(c["traj_foot_duration"].as<float>());
+                step_height_ = IWBC_CHECK(c["step_height"].as<float>());
+                step_length_ = IWBC_CHECK(c["step_length"].as<float>());
+                num_of_cycles_ = IWBC_CHECK(c["num_of_cycles"].as<int>());
```

Comment:
a check on these values could be useful to prevent for example a step lenght to large or a negative step height